akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -1002,6 +1028,8 @@ private[akka] class Shard(
     if (entityId == null || entityId == "") {
       log.warning("{}: Id must not be empty, dropping message [{}]", typeName, msg.getClass.getName)
       context.system.deadLetters ! Dropped(msg, "No recipient entity id", snd, self)
+    } else if (preparingForShutdown && !entities.entityIdExists(entityId)) {
+      log.debug("{}: Not creating entity [{}] as preparing for shutdown [{}]", typeName, entityId, msg.getClass.getName)
```

Comment:
My reasoning here was if these were persistent actors and that would be considered "expensive"

Reply:
We might have to ask what they want. My thinking was that the most important was to have a quick shutdown once the Pods are actually stopped, so that it's possible to have a quick full shutdown-startup bump with as little interruption as possible.

It's good that singletons don't move while preparing for shutdown because that would be abnormal. Also good to not start any rebalance. I'm not so sure about avoiding start of new entities, and maybe even ok to allocate new (previously unknown) shards.

I could have misunderstood what is most important though.

Reply:
^

Reply:
👍 i'll change it to keep sharding working and only speed up shutdown